### DISCUSSION TITLE
GraphQL: How to set the page creation date?

### DISCUSSION THREAD

**Discussion by telkamp** [requarks/wiki#5414] [Help / Questions] [answered]
We are currently converting our MediaWiki content to a new Wikijs installation.
Our current approach is a Python script, reading the full MediaWiki dump (XML), converting each single page version to markdown (using Pandoc) and storing the page versions to Wikijs using the GraphQL API (using `PageMutation` `create` for the first page version and `update` for the subsequent page versions). 
This works so far, even for the page history.
But what we currently not see is the correct date, time and creator/updater of the page versions, so every page history looks like this:
![grafik](https://user-images.githubusercontent.com/13854808/174338141-4ac278bc-b177-46e8-9375-18e15a1f89f5.png)

Question: How can we set the dates and users in the page history? `create` of the `PageMutation` does not implement the `createdAt` and `updatedAt` parameters...

Thank you!

**Comment by NGPixel** [comment, marked as answer]
These dates (and the author) cannot be set by the user because of potential abuse (e.g. a user with sufficient knowledge of the API could send bogus dates and mess up the history). Same issue with letting the user impersonate someone else.

Your only option is to modify the values directly in the DB.

A future API could allow for greater control via restricted elevated permissions.

**Reply by telkamp** [reply]
Many thanks for your answer! Writing directly to the DB should be feasible.
